Add a test

alexey-milovidov opened the pull request:
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

alexey-milovidov committed:
Add a test

alexey-milovidov committed:
Add a test

pamarcos committed:
Merge branch 'master' into fix-bugs

pamarcos committed:
Fix test

pamarcos commented:
Apparently, the test cannot run in parallel. I'll see if I can figure out how to modify it to allow running it in parallel or alternatively add a proper explanation why it cannot

pamarcos committed:
Move test to shell script to fix result_overflow_mode = 'throw' check

ClickHouse does not guarantee that when result_overflow_mode = 'throw'
is used, the rows reported are exactly the maximum set.
result_overflow_mode = 'break' does guarantee that, though.

pamarcos committed:
Add new line at EOF

pamarcos commented:
[`578469c` (#77882)](https://github.com/ClickHouse/ClickHouse/pull/77882/commits/578469cbb5e0dade4894028f5c4483353f66da24) should do it. Turns out that ClickHouse does not guarantee when `result_overflow_mode = 'throw'` the amount of rows reported. Thus, I converted the test to be a shell script one and made some minor modifications to comply with that.

alexey-milovidov commented:
@pamarcos, wow, thank you so much for taking care of this PR! ❤️
